src/main/resources/META-INF/plexus/components.xml

```diff
@@ -18,7 +18,8 @@
                                 org.apache.maven.plugins:maven-resources-plugin:resources
                             </process-resources>
                             <compile>
-                                org.apache.maven.plugins:maven-compiler-plugin:compile
+                                org.apache.maven.plugins:maven-compiler-plugin:compile,
+                                ${project.groupId}:${project.artifactId}:${project.version}:graalvm-resources
```

Comment:
I would bind this goal to the `generate-resources` phase, provided that you don't need the application to be compiled for it to run correctly. If you do need a post-compile phase, then `prepare-package` or `process-classes` might be better options.

Reply:
Thanks Álvaro! 

I tried it first in the `generate-resources` goal, but for Kotlin and Openapi it doesn't work because the openapi yml file and the views are generated during compilation. I'll expore the other options you proposed :)